### PR TITLE
Switching mass hypothis

### PR DESCRIPTION
**Pull request by seneubert**
I added the solution of a DV script that changes a mass hypothesis. Please have a look at code/18-switch-mass-hypo/ntuple_switchHypo.py

I also added a template for the corresponding lessen.

### CODE REVIEW

File: code/18-switch-mass-hypo/ntuple_switchHypo.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+alg.Code = 'ALL' # we could filter out some data here. For now just select ALL candidates
+
+# get the selection(s) created by the stripping
+from PhysSelPython.Wrappers import Selection
+from PhysSelPython.Wrappers import SelectionSequence
+from PhysSelPython.Wrappers import DataOnDemand
+
+strippingSels = [DataOnDemand(Location=tesLoc)] 
+
+# instantiat a Selection that uses the algorithm defined above
```
Comment (betatim) [inline code/18-switch-mass-hypo/ntuple_switchHypo.py]: instantiat -> instantiate

**Comment by betatim** [comment]
From what I can tell the code looks alright. Does it run? (We need #23)

For the lesson we should come up with a good "excuse" why we are doing this, so that people can relate to it. Otherwise it might appears as some advanced gaudi selection kongfu with no purpose.

We should explain the why of the various steps that need to be done. Why a `FilterDesktop` that doesn't seem to do anything, why all these sequences of selection wrappers, what is a `GaudiSequencer`? Or asked differently: how do I know that I need to do these steps? (Otherwise people will just blindly copy the recipe)

**Comment by apuignav** [comment]
I think we should have a lesson before introducing all the concepts. This
will help focusing this session on the mass hypothesis substitution.

**Comment by betatim** [comment]
So #46 is going to be a real thing then. Will you write it?

**Comment by apuignav** [comment]
As soon as I finish LoKi functors, which may take a while if I do it
properly. I will open a PR as soon as I start, so if someone has time to
start before, please feel free to do it.

**Comment by seneubert** [comment]
This code runs. 
You are right, though that for our example Dst -> D0pi it is not very well motivated. I believe for the D0 reco D02pipi is even done in the stripping line already :(
But do we have Dst->D0(pipi)pi MC available?

I have an example where I do D+ --> Lambda_c+ but there is probably no MC for that.

If you know another good use case that would be simple to implement let me know. On the other hand I think there is nothing wrong with putting forward academic examples and let people figure out by themselves when it will become useful.

File: code/18-switch-mass-hypo/ntuple_switchHypo.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from DecayTreeTuple.Configuration import *
+
+# Stream and stripping line we want to use
+stream = 'AllStreams'
+line = 'D2hhCompleteEventPromptDst2D2RSLine'
+tesLoc = '/Event/{0}/Phys/{1}/Particles'.format(stream, line)
+
+# prepare a Filter algorithm to creat a new Selection of particles
+from Configurables import FilterDesktop
+alg = FilterDesktop('SelFilter')
```
Comment (seneubert) [inline code/18-switch-mass-hypo/ntuple_switchHypo.py]: I am not convinced that we need this inital Filter here.
Let me try to simplify this.

**Comment by seneubert** [comment]
The selection wrappers I copied.
I have no Idea why Gaudi works this way. Should I spend time to research this for the lesson?
Can I merge now?

**Pull request merged**